client/java/transports-dataplex/src/main/java/io/openlineage/client/transports/dataplex/DataplexTransport.java

```diff
@@ -0,0 +1,132 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.transports.dataplex;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.datacatalog.lineage.v1.LineageSettings;
+import com.google.cloud.datacatalog.lineage.v1.ProcessOpenLineageRunEventRequest;
+import com.google.cloud.datalineage.producerclient.helpers.OpenLineageHelper;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClientSettings;
+import com.google.protobuf.Struct;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.client.transports.Transport;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DataplexTransport extends Transport implements Closeable {
+
+  private final SyncLineageClient producerClient;
+
+  private final String parent;
+
+  public DataplexTransport(@NonNull DataplexConfig config) throws IOException {
+    this(config, SyncLineageProducerClient.create(createSettings(config)));
+  }
+
+  protected DataplexTransport(@NonNull DataplexConfig config, @NonNull SyncLineageClient client)
+      throws IOException {
+    this.producerClient = client;
+    this.parent =
+        String.format(
+            "projects/%s/locations/%s",
+            getProjectId(config, createSettings(config)),
+            config.getLocations() != null ? config.getLocations() : "us");
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull RunEvent runEvent) {
+    emitEvent(runEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull DatasetEvent datasetEvent) {
+    emitEvent(datasetEvent);
+  }
+
+  @Override
+  public void emit(OpenLineage.@NonNull JobEvent jobEvent) {
+    emitEvent(jobEvent);
+  }
+
+  private <T extends OpenLineage.BaseEvent> void emitEvent(T event) {
+    try {
+      String eventJson = OpenLineageClientUtils.toJson(event);
+      Struct openLineageStruct = OpenLineageHelper.jsonToStruct(eventJson);
+      ProcessOpenLineageRunEventRequest request =
+          ProcessOpenLineageRunEventRequest.newBuilder()
+              .setParent(parent)
+              .setOpenLineage(openLineageStruct)
+              .build();
+      producerClient.processOpenLineageRunEvent(request);
+    } catch (Exception e) {
+      throw new OpenLineageClientException(e);
```

Comment:
It might be good to log the error but not throw an exception. Otherwise this would cause the job to fail.
@midamkina What to you think?

Reply:
I don't think so. Now if the user wants the job to proceed, they can just try-catch this exception. But if we change it to logging and the user wants the job to fail, there will be no straightforward way to do that.

Reply:
Ok I see. So the thing is to make sure that the client integration (e.g. Spark) must properly catch that exception, and then log the error without failing the Spark job.